lms/djangoapps/mobile_api/course_info/views.py

```diff
@@ -166,3 +170,119 @@ def post(self, request, *args, **kwargs):
         # Populate user activity for tracking progress towards a user's course goals
         UserActivity.record_user_activity(user, course_key)
         return Response(status=(200))
+
+
+@view_auth_classes(is_authenticated=False)
+class BlocksInfoInCourseView(BlocksInCourseView):
+    """
+    **Use Case**
+
+        Returns the blocks in the course according to the requesting user's access level.
+        Add to response info fields with information about course
+
+    **Example requests**:
+
+        This api works with all versions {api_version}, you can use: v0.5, v1, v2 or v3
```

Comment:
Why introduce a new version if it can work with previous versions?

Reply:
Hello @jawad-khan, the version 3 was introduced first of all in PR: https://github.com/openedx/edx-platform/pull/33294
Here we have added version 3 support as per comment https://openedx.atlassian.net/wiki/spaces/COMM/pages/3853910042?focusedCommentId=3856859206 for general consistency in new mobile applications.

Ed's suggestion was to make it more intuitive for using mobile API with new and old apps:
> one could assume that any call to the API version n or above would be for the new apps, < n for the legacy apps